### PR TITLE
feat: Add 'Delete Catalog' feature in the catalog details page

### LINKED ISSUE #37: Delete Catalog

### PR DESCRIPTION
**Pull request by robertgv**
Closes #37 

https://github.com/user-attachments/assets/cfde7a34-a2c9-42b8-8037-851567c089a9

Thanks to @jamieknight-db for providing a nice baseline with the delete table PR. 😄

### CODE REVIEW

File: src/components/modals/DeleteCatalogModal.tsx [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+  });
+
+  const handleSubmit = useCallback(() => {
+    mutation.mutate(
+      { name: catalog },
+      {
+        onError: (error: Error) => {
+          setNotification(error.message, 'error');
+        },
+      }
```
Comment (jamieknight-db) [inline src/components/modals/DeleteCatalogModal.tsx:34]: This is nice 👍  we should do the same error handling on all the delete assets.

**Review by jamieknight-db — APPROVED** [review]
LGTM! Thank you, Robert!

**Review by Romanize — APPROVED** [review]

File: src/components/modals/DeleteCatalogModal.tsx [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+      destroyOnClose
+      onCancel={closeModal}
+      onOk={handleSubmit}
+      okButtonProps={{ loading: mutation.isPending }}
+    >
+      <Typography.Text>
+        Are you sure you want to delete the catalog
+      </Typography.Text>
+      <Typography.Text strong>
+        {` ${catalog}`}
```
Comment (Romanize) [inline src/components/modals/DeleteCatalogModal.tsx:58]: nit: you can use `{' '}` for cleaner spacing

**Review by yc-shawn — APPROVED** [review]
Nice!

File: src/components/modals/DeleteCatalogModal.tsx (hunk 2 above)
Comment (robertgv) [reply src/components/modals/DeleteCatalogModal.tsx:58]: Thanks for the tip @Romanize!

**Comment by robertgv** [comment]
Could someone squash and merge this PR? I do not have permissions.
Thank you! 😄

File: src/components/modals/DeleteCatalogModal.tsx (hunk 1 above)
Comment (robertgv) [reply src/components/modals/DeleteCatalogModal.tsx:34]: Thank you @jamieknight-db! 😄

**Pull request merged**